_includes/markdown/JavaScript.md

```diff
@@ -304,6 +304,77 @@ document.getElementById( 'menu' ).addEventListener( 'click', ( e ) => {
 
 You may be wondering why we don't just add one listener to the `<body>` for all our events. Well, we want the event to *bubble up the DOM as little as possible* for [performance reasons](https://jsperf.com/event-delegation-distance). This would also be pretty messy code to write.
 
+### Debounce and Throttle
+
+Browser events such as scrolling, resizing, and cursor movements happen as fast as possible and can cause performance issues. Debouncing and throttling our functions helps us increase performance by controlling the rate at which an event listener calls them. 
+
+Debouncing a function will prevent it from being called again until a defined amount of time has passed, i.e., execute this function if 200ms has passed since it was last called. A common use case would be when resizing a browser window; we can apply classes or move elements after the resize has happened.
```

Comment:
Let's link to a debouncing documentation resource

Reply:
Linked to underscore, lodash, and CSS Tricks explanations at the end. Let me know if you were thinking something else.